internal/opa/server.go

```diff
@@ -0,0 +1,127 @@
+package opa
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	vspheremodel "github.com/kubev2v/forklift/pkg/controller/provider/model/vsphere"
+	"github.com/kubev2v/migration-planner/internal/agent/collector"
+	"github.com/open-policy-agent/opa/runtime"
+	"go.uber.org/zap"
+)
+
+// Server handles the OPA runtime server
+type Server struct {
+	config  *Config
+	runtime *runtime.Runtime
+	ctx     context.Context
+	cancel  context.CancelFunc
+}
+
+// NewServer creates a new OPA server instance
+func NewServer(config *Config) *Server {
+	ctx, cancel := context.WithCancel(context.Background())
+	return &Server{
+		config: config,
+		ctx:    ctx,
+		cancel: cancel,
+	}
+}
+
+// Initialize starts the OPA runtime server
+func (s *Server) Initialize() error {
+	// Start OPA runtime server
+	if err := s.startOPARuntime(); err != nil {
+		return fmt.Errorf("failed to start OPA runtime: %w", err)
+	}
+
+	// Wait for OPA server to be ready
+	if err := s.waitForServer(); err != nil {
```

Comment:
in a loaded env, there is no guarantee that the server will come up under 1s. so If the server is up by the time you make the first request, this will fail.